packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -52,6 +52,19 @@ class TextSelectionPoint {
   /// Direction of the text at this edge of the selection.
   final TextDirection? direction;
 
+  @override
+  bool operator ==(Object other) {
```

Comment:
This is a bug I found that the selectionEndPoint setter does the list comparison. which will always returns false because it does identity compare.

Reply:
Great catch!  It looks like you're right.